core/services/vrf/v2/listener_v2_log_listener.go

```diff
@@ -0,0 +1,451 @@
+		case <-lsn.chStop:
+			return
+		case <-ticker.C:
+			start := time.Now()
+			lsn.l.Debugw("log listener loop")
+			// Filter registration is idempotent, so we can just call it every time
+			// and retry on errors using the ticker.
+			err := lsn.chain.LogPoller().RegisterFilter(logpoller.Filter{
+				Name: fmt.Sprintf("vrf_%s_keyhash_%s_job_%d", lsn.coordinator.Version(), lsn.job.VRFSpec.PublicKey.MustHash().String(), lsn.job.ID),
+				EventSigs: evmtypes.HashArray{
+					lsn.coordinator.RandomWordsFulfilledTopic(),
+					lsn.coordinator.RandomWordsRequestedTopic(),
+				},
+				Addresses: evmtypes.AddressArray{
+					lsn.coordinator.Address(),
+				},
+			})
+			if err != nil {
+				lsn.l.Errorw("error registering filter in log poller, retrying",
+					"err", err,
+					"elapsed", time.Since(start))
+				continue
+			}
+
+			// on startup we want to initialize the last processed block
+			if startingUp {
+				lsn.l.Debugw("initializing last processed block on startup")
+				lastProcessedBlock, err = lsn.initializeLastProcessedBlock(ctx)
+				if err != nil {
+					lsn.l.Errorw("error initializing last processed block, retrying",
+						"err", err,
+						"elapsed", time.Since(start))
+					continue
+				}
+				startingUp = false
+				lsn.l.Debugw("initialized last processed block", "lastProcessedBlock", lastProcessedBlock)
+			}
+
+			pending, err := lsn.pollLogs(ctx, minConfs, lastProcessedBlock)
+			if err != nil {
+				lsn.l.Errorw("error polling vrf logs, retrying",
+					"err", err,
+					"elapsed", time.Since(start))
+				continue
+			}
+
+			// process pending requests and insert any fulfillments into the inflight cache
+			lsn.processPendingVRFRequests(ctx, pending)
+
+			lastProcessedBlock, err = lsn.updateLastProcessedBlock(ctx, lastProcessedBlock)
+			if err != nil {
+				lsn.l.Errorw("error updating last processed block, continuing anyway", "err", err)
+			} else {
+				lsn.l.Debugw("updated last processed block", "lastProcessedBlock", lastProcessedBlock)
+			}
+			lsn.l.Debugw("log listener loop done", "elapsed", time.Since(start))
+		}
+	}
+}
+
+// initializeLastProcessedBlock returns the earliest block number that we need to
+// process requests for. This is the block number of the earliest unfulfilled request
+// or the latest finalized block, if there are no unfulfilled requests.
+// TODO: add tests
+func (lsn *listenerV2) initializeLastProcessedBlock(ctx context.Context) (lastProcessedBlock int64, err error) {
+	lp := lsn.chain.LogPoller()
+	start := time.Now()
+
+	// will retry on error in the runLogListener loop
+	latestBlock, err := lp.LatestBlock()
+	if err != nil {
+		return 0, errors.Wrap(err, "LogPoller.LatestBlock()")
+	}
+	fromTimestamp := time.Now().UTC().Add(-lsn.job.VRFSpec.RequestTimeout)
+	ll := lsn.l.With(
+		"latestFinalizedBlock", latestBlock.FinalizedBlockNumber,
+		"latestBlock", latestBlock.BlockNumber,
+		"fromTimestamp", fromTimestamp)
+	ll.Debugw("Initializing last processed block")
+	defer func() {
```

Comment:
random and total aside: if we had tracing (which we are working for loopps) would you still need this kind of logging?

Reply:
Probably not, tracing can replace all the "elapsed" style logging we do.